### PR TITLE
AFHS-388 Added missing test confirming we'll let you know email sent when address mismatched

### PR DESCRIPTION
**Pull request by YetAnotherMatt**

### CODE REVIEW

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java [hunk 1]
```diff
@@ -68,6 +70,25 @@ void shouldRequestNewCardAndSendEmailForSuccessfulClaim() throws JsonProcessingE
         wiremockManager.assertThatDepositFundsRequestMadeForPayment(payment);
     }
 
+    @Test
+    void shouldSendWeWillLetYouKnowEmailForAddressMismatch() throws JsonProcessingException, NotificationClientException {
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java]: Did you also want to check that the letter got sent here too?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java [hunk 2]
```diff
@@ -68,6 +70,25 @@ void shouldRequestNewCardAndSendEmailForSuccessfulClaim() throws JsonProcessingE
...
+        wiremockManager.stubEligibilityResponse(anIdMatchedEligibilityConfirmedAddressNotMatchedResponse());
+        stubNotificationEmailResponse();
+
+        ResponseEntity<ClaimResultDTO> response = restTemplate.exchange(buildClaimRequestEntity(claimDTO), ClaimResultDTO.class);
+        invokeAllSchedulers();
+
+        assertThat(response.getStatusCode()).isEqualTo(OK);
+        assertThat(response.getBody().getClaimStatus()).isEqualTo(ClaimStatus.REJECTED);
+        Claim claim = repositoryMediator.getClaimForNino(claimDTO.getClaimant().getNino());
+        assertThat(claim.getClaimStatus()).isEqualTo(ClaimStatus.REJECTED);
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java]: Do we need to check the claim status both on the response and in the database?

Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java]: We do that in other tests too. I guess we don't need to here as well.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java (hunk 1 above)
Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java]: Added a check for the letter being sent

**Pull request merged**